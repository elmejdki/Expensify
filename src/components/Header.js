import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <div className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/dashboard">
          <h1>Exspensify</h1>
        </Link>
        <button
          className="button button--link"
          onClick={startLogout}
        >
          Log Out
        </button>
      </div>
    </div>
  </div>
);

const mapDispatchToProps = { startLogout };

export default connect(undefined, mapDispatchToProps)(Header);