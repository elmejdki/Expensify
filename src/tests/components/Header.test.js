import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import Header from '../../components/Header';

const renderer = new ShallowRenderer();

it('should render Header correctly', () => {
  renderer.render(<Header />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
