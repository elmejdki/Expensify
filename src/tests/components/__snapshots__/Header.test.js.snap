// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<div
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        to="/dashboard"
      >
        <h1>
          Exspensify
        </h1>
      </Link>
      <button
        className="button button--link"
      >
        Log Out
      </button>
    </div>
  </div>
</div>
`;
